vm-device/src/lib.rs

```diff
@@ -1,3 +1,14 @@
+/// Errors associated with VM management
+#[derive(Debug)]
+pub enum Error {
+    /// Could not pause a migratable.
+    Pause,
```

Comment:
The problem with this kind of error defined from the crate itself, it's not flexible enough to propagate the original error.
Let's say you have a `VirtioRng` which implements `Migratable`, it will have to return either `Pause` or `Resume` if an error of any sort comes up. But you want to keep the original error that cause the pause or the resume to fail.
I think the way to do it is to define a generic error so that an `io::Error` can be passed.